mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -107,7 +107,7 @@ private[regression] trait GeneralizedLinearRegressionBase extends PredictorParam
         s"with ${$(family)} family does not support ${$(link)} link function.")
     }
     val newSchema = super.validateAndTransformSchema(schema, fitting, featuresDataType)
-    if ($(linkPredictionCol).nonEmpty) {
+    if (isDefined(linkPredictionCol) && $(linkPredictionCol).nonEmpty) {
```

Comment:
This is used twice now, perhaps makes sense to make a `def hasLinkPredictionCol` as for e.g. `hasQuantilesCol`?